examples/heat_equation.jl

```diff
@@ -0,0 +1,49 @@
+"""
+Solve the heat equation ∂ₜu = Δu on [0,1] using OrdinaryDiffEq for time stepping.
+Boundary conditions are such that an analytic solution is given by:
+    sin(2π*x) * exp(-(2*π)^2*t)
+
+This example demonstrates how to solve time dependent PDE's combining `fenics` and `DiffEq`.
+
+"""
+module HeatEquationExample
+
+using Test
+using OrdinaryDiffEq
+using FEniCS
+
+mesh = UnitIntervalMesh(20)
+V = FunctionSpace(mesh,"P",1)
+u_str = "sin(2*pi*x[0]) * exp(-t*pow(2*pi, 2))"
+
+u = interpolate(Expression(u_str, degree=2, pi=Float64(pi), t=0), V)
+
+bc = DirichletBC(V, 0.0, "on_boundary")
+dudt = TrialFunction(V)
+v = TestFunction(V)
+Dudt = FeFunction(V)
+F = dot(dudt, v)*dx + dot(grad(u), grad(v))*dx
+a = lhs(F)
+L = rhs(F)
+p = (u=u, Dudt=Dudt, bc=bc, a=a, L=L)
+
+function f!(dudt_vec, u_vec, p, t)
+    assign(p.u, u_vec)
+    lvsolve(p.a, p.L, p.Dudt, p.bc)
+    copy!(dudt_vec, get_array(p.Dudt))
+end
+
+u0_vec = get_array(u)
+t_final = 0.1
+tspan = (0.0, t_final)
+prob = ODEProblem(f!, u0_vec, tspan, p)
+u_true = interpolate(Expression(u_str, degree=2, pi=Float64(pi), t=t_final), V)
+
+alg = ImplicitEuler(autodiff=false)
+sol = OrdinaryDiffEq.solve(prob, alg)
```

Comment:
```suggestion
sol = OrdinaryDiffEq.solve(prob, alg, reltol=1e-6)
```

? It might just be a tolerance thing.

Reply:
Note that `ImplicitEuler` is adaptive

Reply:
If I change `reltol=1e-6` and refine the grid to 50 I get an error of 0.2%. 
If I change `reltol=1e-6` and refine the grid to 1000 I get an error of 0.7%.
Is it expected that refining the grid can increase error in this setup?

Reply:
not really, but with that tolerance there's a random of errors you'd expect. Drop it even lower then and it should go down more: `reltol=1e-9,abstol=1e-9`. ImplicitEuler is really slow though, so you might want to use `Rosenbrock23(autodiff=false)`, `Rodas5(autodiff=false)`, or `KenCarp4(autodiff=false)` when you get to that.

Reply:
Ok thanks! It's possible to get the error small this way. I go with 1% error to keep the example fast.